android/src/main/kotlin/com/onegini/mobile/sdk/flutter/useCases/AuthenticateUserImplicitlyUseCase.kt

```diff
@@ -12,15 +12,18 @@ import io.flutter.plugin.common.MethodChannel
 
 class AuthenticateUserImplicitlyUseCase(private var oneginiClient: OneginiClient) {
     operator fun invoke(call: MethodCall, result: MethodChannel.Result) {
-        val scope = call.argument<ArrayList<String>>("scope")
-        val userProfile = oneginiClient.userClient.userProfiles.firstOrNull()
+        val profileId = call.argument<String>("profileId")
+        val scopes = call.argument<ArrayList<String>>("scopes")
+
+        val userProfile = oneginiClient.userClient.userProfiles.find { it.profileId == profileId }
+
         if (userProfile == null) {
```

Comment:
If we're modifying this method already, let's also make it throw a correct error or let's make a story to rework all of them. 
`USER_PROFILE_IS_NULL` is definitely not descriptive to the users of our wrapper. It probably should be, something like `PROFILE_DOES_NOT_EXIST`. The exact name is up to you, but this is the error enum we use in RN.

Reply:
updated